diff --git a/viewer/vueapp/src/components/help/Help.vue b/viewer/vueapp/src/components/help/Help.vue
index ba43003a..89b292f0 100644
--- a/viewer/vueapp/src/components/help/Help.vue
+++ b/viewer/vueapp/src/components/help/Help.vue
@@ -148,6 +148,9 @@
           </h3>
           <div class="row">
             <div class="col-sm-12">
+              <a class="btn btn-link" href="https://github.com/idaholab/Malcolm">Malcolm on GitHub</a> |
+              <a class="btn btn-link" href="/readme">Malcolm README</a> |
+              <a class="btn btn-link" href="/cyberchef/">CyberChef</a> |
               <a class="btn btn-link" href="https://molo.ch">Home Page</a> |
               <a class="btn btn-link" href="https://molo.ch/faq">FAQ</a> |
               <a class="btn btn-link" href="https://github.com/aol/moloch/wiki">Wiki</a> |
