From 23bdf6c852359b8677bca61805126faab6a19b14 Mon Sep 17 00:00:00 2001
From: Andy Wick <andywick@gmail.com>
Date: Mon, 3 Feb 2020 13:04:40 -0500
Subject: [PATCH] don't actually hunt sessions without fileId set fixes #1374

---
 CHANGELOG        | 5 +++++
 viewer/viewer.js | 7 ++++++-
 2 files changed, 11 insertions(+), 1 deletion(-)

diff --git a/CHANGELOG b/CHANGELOG
index 46bbc77d..988f6e5c 100644
--- a/CHANGELOG
+++ b/CHANGELOG
@@ -18,6 +18,11 @@ Node Versions:
 NOTICE: Restart wiseService before capture when upgrading
 
 2.3.0 2020/02/xx
+  - s3 - Fixes the problem where the s3 token expires during a capture (issue #1370)
+  - viewer - Fix decode crash (thanks mammo0)
+  - capture - New tcphealthcheck plugin (thanks fj604)
+  - viewer - initial pie chart
+  - viewer - fix viewer crash when hunting fake sessions (issue #1374)
 
 2.2.1 2020/01/21
   - capture - fix --skip not working with ES 7.x
diff --git a/viewer/viewer.js b/viewer/viewer.js
index 176b249f..12e87a09 100644
--- a/viewer/viewer.js
+++ b/viewer/viewer.js
@@ -7210,6 +7210,11 @@ function runHuntJob (huntId, hunt, query, user) {
       let sessionId = Db.session2Sid(hit);
       let node = session.node;
 
+      // There is no files, this is a fake session, don't hunt it
+      if (session.fileId === undefined || session.fileId.length === 0) {
+          return updateHuntStats(hunt, huntId, session, searchedSessions, cb);
+      }
+
       isLocalView(node, function () {
         sessionHunt(sessionId, options, function (err, matched) {
           if (err) {
@@ -7347,7 +7352,7 @@ function processHuntJob (huntId, hunt) {
             }
           };
 
-          query._source = ['lastPacket', 'node', 'huntId', 'huntName'];
+          query._source = ['lastPacket', 'node', 'huntId', 'huntName', 'fileId'];
 
           if (Config.debug > 2) {
             console.log('HUNT', hunt.name, hunt.userId, '- start:', new Date(hunt.lastPacketTime || hunt.query.startTime * 1000), 'stop:', new Date(hunt.query.stopTime * 1000));
-- 
2.20.1

