diff --git a/netbox/extras/constants.py b/netbox/extras/constants.py
index fadf59c..16563ea 100644
--- a/netbox/extras/constants.py
+++ b/netbox/extras/constants.py
@@ -40,20 +40,6 @@ DEFAULT_DASHBOARD = [
             ]
         }
     },
-    {
-        'widget': 'extras.NoteWidget',
-        'width': 4,
-        'height': 2,
-        'title': 'Welcome!',
-        'color': 'green',
-        'config': {
-            'content': (
-                'This is your personal dashboard. Feel free to customize it by rearranging, resizing, or removing '
-                'widgets. You can also add new widgets using the "add widget" button below. Any changes affect only '
-                '_your_ dashboard, so feel free to experiment!'
-            )
-        }
-    },
     {
         'widget': 'extras.ObjectCountsWidget',
         'width': 4,
@@ -70,32 +56,6 @@ DEFAULT_DASHBOARD = [
             ]
         }
     },
-    {
-        'widget': 'extras.RSSFeedWidget',
-        'width': 4,
-        'height': 4,
-        'title': 'NetBox News',
-        'config': {
-            'feed_url': 'https://api.netbox.oss.netboxlabs.com/v1/newsfeed/',
-            'max_entries': 10,
-            'cache_timeout': 14400,
-            'requires_internet': True,
-        }
-    },
-    {
-        'widget': 'extras.ObjectCountsWidget',
-        'width': 4,
-        'height': 3,
-        'title': 'Circuits',
-        'config': {
-            'models': [
-                'circuits.provider',
-                'circuits.circuit',
-                'circuits.providernetwork',
-                'circuits.provideraccount',
-            ]
-        }
-    },
     {
         'widget': 'extras.ObjectCountsWidget',
         'width': 4,
