diff --git a/viewer/db.js b/viewer/db.js
index 71470bb1..3c00c65a 100644
--- a/viewer/db.js
+++ b/viewer/db.js
@@ -343,6 +343,8 @@ function fixSessionFields (fields, unflatten) {
   if (unflatten) {
     fields.source = { as: {}, geo: {} };
     fields.destination = { as: {}, geo: {} };
+    fields.client = {};
+    fields.server = {};
   }
   for (const f in fields) {
     const path = f.split('.');
