diff --git a/capture/moloch.h b/capture/moloch.h
index 06a79110..5dd5da6e 100644
--- a/capture/moloch.h
+++ b/capture/moloch.h
@@ -352,7 +352,7 @@ enum MolochRotate {
  */
 
 #define MOLOCH_FIELD_NOT_FOUND  -1
-#define MOLOCH_FIELDS_DB_MAX 512
+#define MOLOCH_FIELDS_DB_MAX 2048
 #define MOLOCH_FIELDS_CNT_MIN MOLOCH_FIELDS_DB_MAX
 #define MOLOCH_FIELDS_CNT_MAX (MOLOCH_FIELDS_DB_MAX*2)
 #define MOLOCH_FIELD_EXSPECIAL_START            (MOLOCH_FIELDS_CNT_MAX)
diff --git a/capture/plugins/wise.c b/capture/plugins/wise.c
index f311d743..eb4ba1e0 100644
--- a/capture/plugins/wise.c
+++ b/capture/plugins/wise.c
@@ -111,7 +111,7 @@ typedef struct wiseitem_head {
     uint32_t              wil_count;
 } WiseItemHead_t;
 
-#define WISE_MAX_REQUEST_ITEMS 512
+#define WISE_MAX_REQUEST_ITEMS MOLOCH_FIELDS_DB_MAX
 typedef struct wiserequest {
     BSB          bsb;
     WiseItem_t  *items[WISE_MAX_REQUEST_ITEMS];
