diff --git a/CHANGELOG b/CHANGELOG
index b758055d..ee4be26c 100644
--- a/CHANGELOG
+++ b/CHANGELOG
@@ -22,6 +22,10 @@ Node Versions:
 NOTICE: Restart wiseService before capture when upgrading
 NOTICE: Cross-cluster Shortcuts require you to not restart all your viewers at once after upgrading
 
+=======
+3.2.1 2022/01/xx
+  - viewer - fix --insecure which broke in 3.2.0
+
 3.2.0 2021/12/07
   - release - node 14.18.2
   - release - remove daily.sh, setup a cron directly now
diff --git a/viewer/db.js b/viewer/db.js
index 3a937c03..4c471331 100644
--- a/viewer/db.js
+++ b/viewer/db.js
@@ -133,6 +133,9 @@ exports.initialize = async (info, cb) => {
 
   if (info.usersHost) {
     User.initialize({
+      insecure: info.insecure,
+      ca: info.ca,
+      requestTimeout: info.requestTimeout,
       node: info.usersHost,
       clientKey: info.esClientKey,
       clientCert: info.esClientCert,
@@ -144,6 +147,9 @@ exports.initialize = async (info, cb) => {
     });
   } else {
     User.initialize({
+      insecure: info.insecure,
+      ca: info.ca,
+      requestTimeout: info.requestTimeout,
       node: info.host,
       clientKey: info.esClientKey,
       clientCert: info.esClientCert,
diff --git a/wiseService/wiseService.js b/wiseService/wiseService.js
index 6faa3228..9f9ef783 100644
--- a/wiseService/wiseService.js
+++ b/wiseService/wiseService.js
@@ -224,6 +224,7 @@ function setupAuth () {
   const es = getConfig('wiseService', 'usersElasticsearch', 'http://localhost:9200');
 
   User.initialize({
+    insecure: internals.insecure,
     node: es,
     prefix: getConfig('wiseService', 'usersPrefix', ''),
     apiKey: getConfig('wiseService', 'usersElasticsearchAPIKey'),
