diff --git a/viewer/vueapp/src/components/utils/Footer.vue b/viewer/vueapp/src/components/utils/Footer.vue
index 50e30c99..d54542dc 100644
--- a/viewer/vueapp/src/components/utils/Footer.vue
+++ b/viewer/vueapp/src/components/utils/Footer.vue
@@ -5,6 +5,8 @@
       <small>
         Arkime v{{molochVersion}} |
         <a href="https://arkime.com">arkime.com</a>
+        | <a href="/readme">Malcolm</a>
+        | <a href="/dashboards">Dashboards</a>
         <span v-if="responseTime && !loadingData">
           | {{ responseTime | commaString }}ms
         </span>
