diff --git a/viewer/vueapp/src/components/spiview/Spiview.vue b/viewer/vueapp/src/components/spiview/Spiview.vue
index 76dd613b..b8117034 100644
--- a/viewer/vueapp/src/components/spiview/Spiview.vue
+++ b/viewer/vueapp/src/components/spiview/Spiview.vue
@@ -960,6 +960,8 @@ export default {
 
       // sorted list of categories for the view
       this.categoryList = Object.keys(this.categoryObjects).sort();
+      this.categoryList.splice(this.categoryList.indexOf('malcolm'), 1);
+      this.categoryList.unshift('malcolm');
       this.categoryList.splice(this.categoryList.indexOf('general'), 1);
       this.categoryList.unshift('general');
 
