diff --git a/viewer/vueapp/src/components/spiview/Spiview.vue b/viewer/vueapp/src/components/spiview/Spiview.vue
index 66e08ecf..f7156b80 100644
--- a/viewer/vueapp/src/components/spiview/Spiview.vue
+++ b/viewer/vueapp/src/components/spiview/Spiview.vue
@@ -915,6 +915,8 @@ export default {

       // sorted list of categories for the view
       this.categoryList = Object.keys(this.categoryObjects).sort();
+      this.categoryList.splice(this.categoryList.indexOf('malcolm'), 1);
+      this.categoryList.unshift('malcolm');
       this.categoryList.splice(this.categoryList.indexOf('general'), 1);
       this.categoryList.unshift('general');

