diff --git a/viewer/vueapp/src/components/sessions/SessionDetail.vue b/viewer/vueapp/src/components/sessions/SessionDetail.vue
index b2f32a02..f0459680 100644
--- a/viewer/vueapp/src/components/sessions/SessionDetail.vue
+++ b/viewer/vueapp/src/components/sessions/SessionDetail.vue
@@ -846,7 +846,7 @@ export default {
 .session-detail dt {
   float: left;
   clear: left;
-  width: 160px;
+  width: 320px;
   text-align: right;
   margin-right: 6px;
   line-height: 1.7;
@@ -854,7 +854,7 @@ export default {
 }
 
 .session-detail dd {
-  margin-left: 165px;
+  margin-left: 325px;
 }
 
 /* more items link */
