diff --git a/viewer/db.js b/viewer/db.js
index 96a75ecd..0289b67c 100644
--- a/viewer/db.js
+++ b/viewer/db.js
@@ -392,7 +392,13 @@ function fixSessionFields (fields, unflatten) {
     delete fields[f];
     for (let i = 0; i < path.length; i++) {
       if (i === path.length - 1) {
-        key[path[i]] = value;
+        // HACKITY-HACK-HACK
+        // This was breaking in OpenSearch v2. @andywick says he doesn't think the .keyword
+        // fields should be showing up here, but here we are.
+        // "Doctor, it hurts when I do this." "Don't do that."
+        if (path[i] !== 'keyword') {
+          key[path[i]] = value;
+        }
         break;
       } else if (key[path[i]] === undefined) {
         key[path[i]] = {};
