diff -Naur a/viewer/viewer.js b/viewer/viewer.js
--- a/viewer/viewer.js	2018-10-03 13:08:06.228039893 -0600
+++ b/viewer/viewer.js	2018-10-03 13:10:44.327829524 -0600
@@ -221,8 +221,11 @@
     }
 
     // Header auth
+    console.log("DEBUG 1 - checking userNameHeader of", internals.userNameHeader, "in", req.headers, "for", req.url);
     if (internals.userNameHeader !== undefined) {
+      console.log("DEBUG 2 - checking userNameHeader of", internals.userNameHeader, "in", req.headers, "for", req.url);
       if (req.headers[internals.userNameHeader] !== undefined) {
+        console.log("DEBUG 3 - checking userNameHeader of", internals.userNameHeader, "in", req.headers, "for", req.url);
         var userName = req.headers[internals.userNameHeader];
         Db.getUserCache(userName, function(err, suser) {
           if (err) {return res.send("ERROR - getUser - user: " + userName + " err:" + err);}
@@ -239,6 +242,7 @@
         console.log("DEBUG - Couldn't find userNameHeader of", internals.userNameHeader, "in", req.headers, "for", req.url);
       }
     }
+    console.log("DEBUG 4 - after userNameHeader of", internals.userNameHeader, "in", req.headers, "for", req.url);
 
     // Browser auth
     req.url = req.url.replace("/", Config.basePath());
